### PR TITLE
Fixing issue caused by azure-common not accepting profile and api_ver…

### PR DESCRIPTION
**Pull request by zikalino**
…sion at the same time

##### SUMMARY
This is to fix issue which is not yet detected by Ansible CI.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
azure_rm_containerregistry

##### ANSIBLE VERSION
2.5

### CODE REVIEW

File: lib/ansible/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -942,7 +942,7 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None):
         # If the client doesn't accept api_version, it's unversioned.
         # If it does, favor explicitly-specified api_version, fall back to api_profile
         if 'api_version' in client_argspec.args:
-            profile_default_version = api_profile_dict.get('default_api_version', None)
+            profile_default_version = api_profile_dict.get('default_api_version', None) if api_profile_dict else None
```
Comment (yungezz) [inline lib/ansible/module_utils/azure_rm_common.py]: is the default  profile_default_version still necessary?

Comment (yungezz) [reply lib/ansible/module_utils/azure_rm_common.py]: why not just remove it, use api_version or not.

Comment (zikalino) [reply lib/ansible/module_utils/azure_rm_common.py]: i wonder what should take precedence, if profile is there, and module specifies api_version, we should maybe set profile to none

**Comment by wilsonge** [comment]
Confirmed that this patch works successfully. This can be tested by trying to run the `azure_rm_containerregistry` module which currently fails with the aforementioned error in v2.5.3. It works after this patch

**Comment by zikalino** [comment]
seems like this issue was solved in a meanwhile by @nitzmahone , so i am closing this pr

**Comment by wilsonge** [comment]
If that's the versioning pin'ing fix can we make sure that get's backported into the 2.5 series?

**Pull request closed without merging**